### PR TITLE
Publish public methods

### PR DESCRIPTION
**Pull request by n1313**
This PR brings the changes from https://github.com/styleguidist/react-styleguidist/pull/280 into `next` branch.

### CODE REVIEW

File: loaders/props-loader.js [hunk 1]
```diff
@@ -33,6 +33,15 @@ module.exports = function(source) {
 		console.log();
 	}
 
+	if (parsedProps.length) {
```
Comment (sapegin) [inline loaders/props-loader.js]: This check is unnecessary, forEach will do nothing for an empty array.

File: examples/basic/lib/components/CounterButton/CounterButton.css [hunk 2]
```diff
@@ -0,0 +1,10 @@
+.button {
```
Comment (sapegin) [inline examples/basic/lib/components/CounterButton/CounterButton.css]: We don’t use CSS Modules here and already have a `.button` class in `Button` component.

File: src/rsg-components/ReactComponent/ReactComponentRenderer.js [hunk 3]
```diff
@@ -50,6 +50,9 @@ const styles = ({ font, monospace, light }) => ({
 	props: {
 		marginBottom: 30,
 	},
+	methods: {
```
Comment (sapegin) [inline src/rsg-components/ReactComponent/ReactComponentRenderer.js]: Could be `.section` or something like that, and would work for both props and methods.

**Comment by sapegin** [comment]
Could you please also add tests for the loader?

File: loaders/props-loader.js (hunk 1 above)
Comment (n1313) [inline loaders/props-loader.js]: Without this check the test for props-loader fails. I don't understand the reason, but in test env `parsedProps` is an object, and in dev/prod env it's an array of objects. I was going to ask you about it in chat but it was late so I just slapped a check on it and went to bed -___-

**Comment by n1313** [comment]
Yes, this is a copy of code from master with minor changes.

File: loaders/props-loader.js (hunk 1 above)
Comment (sapegin) [inline loaders/props-loader.js]: OK, I’ll take a look later.

File: src/rsg-components/ReactComponent/ReactComponentRenderer.js (hunk 3 above)
Comment (sapegin) [inline src/rsg-components/ReactComponent/ReactComponentRenderer.js]: For some reason react-docs’ `parse` can return either array or object: 
https://github.com/reactjs/react-docgen/blob/master/src/parse.js#L52

And it’a actually expected, see `castArray` below your code. We just need to move it upper.

You can cherry-pick this commit: https://github.com/styleguidist/react-styleguidist/commit/d9c933bad017f2d4e629f0cc34f6d26369004dfe

Comment (sapegin) [inline src/rsg-components/ReactComponent/ReactComponentRenderer.js]: And this one: 53df991b4f1bc8d9d037ed70c3bb388ea8bf0c3a

**Comment by n1313** [comment]
I've updated the PR. Please take a look!

**Comment by sapegin** [comment]
Awesome! Can you add some tests and I’ll merge it?

**Pull request merged**